### ISSUE TITLE
override onQueryProgress vs. StreamingQueryListenerHandlers

### ISSUE THREAD

**Issue by dougspadotto** [mspnp/spark-monitoring#196]
Hello.
If I use a custom onQueryProgress override, will it keep this component from sending data onwards to LogAnalytics? 

My situations is that for now I also need to send some observables to Kafka as well as send all Spark data to LogAnalytics using this component, and I'm seeing a behavior where whenever I add any code to override onQueryProgress, it doesn't seem to be sending data over to LogAnalytics.

**Comment by hallihan** [comment]
I haven't seen this scenario before, are you adding a new listener like in https://spark.apache.org/docs/3.2.0/structured-streaming-programming-guide.html#reporting-metrics-programmatically-using-asynchronous-apis ?

**Comment by dougspadotto** [comment]
Yes, I am seeing org.apache.spark.sql.streaming.StreamingQueryListener$QueryStartedEvent sent over, but no QueryProgress events when also sending to Kafka the information by overriding onQueryProgress.

Our onQueryProgress override looks like this:

```
    override def onQueryProgress(queryProgress: QueryProgressEvent): Unit = {
       ...
      val queryJson = queryProgress.progress.toString
      
      val jsonLogs = s"""{
                        "id_sdd": "EQ2",
       ...
                        "query_progress": $queryJson
                      }"""
   
      spark
          .read
          .json(Seq(jsonLogs).toDS)
          .select(...))
          .write
          .format("kafka")
         ...
          .save() 
}
```

**Comment by hallihan** [comment]
Just confirming, this is part of your own class that extends SparkListener, and you're adding that listener via addListener directly or via the spark.extraListeners config? I'm just confirming that this code is separate from, and not being added to the spark-monitoring code.

**Comment by dougspadotto** [comment]
We're adding via addListener directly.

**Comment by dougspadotto** [comment]
I think the issue was not the override after all, @hallihan .

I tested the same code sending data to a brand new LogAnalytics workspace and it went through without issue. 

I did that because I saw several warnings on the Operational Issues section of the original LogAnalytics workspace.
One that kept being repeated was "Data of type SparkListenerEvent is being dropped due to incorrect format at lineOffset: -1. Exception message: The reader's MaxDepth of 64 has been exceeded.". 

I believe this warning was the culprit for the data not coming through. But I'm not sure if this is this is due to this library's code or LogAnalytics inability to receive more data once this warning appears. Could it be that the LogAnalytics' table's schema is broken and won't allow for more data to come through?

I've attached here the list of warnings received.
[export_data.xlsx](https://github.com/mspnp/spark-monitoring/files/9685001/export_data.xlsx)

**Comment by hallihan** [comment]
Thanks for the additional data.  I was just getting ready to point out that the implementation at https://github.com/apache/spark/blob/v3.2.0/core/src/main/scala/org/apache/spark/util/ListenerBus.scala#L101 seems pretty clear that all listeners should receive the same data. We know there are sometimes issues where LogAnalytics does not process records (field count, field size, etc) but the spark-monitoring library doesn't have much control here.  It passes through the events just as they are generated by Spark or the workload, and if these exceed the limits of Log Analytics, then warnings and errors are expected.  The complexity of the events generated by Spark and and workloads in the Spark ecosystem has grown considerably over time. I'll take a look at the warnings and see if I can find any reasoning for the workspace getting into a bad state. If you see this new workspace hit a similar state, please update this thread with any additional clues.

**Comment by hallihan** [comment]
Closing out since additional data for troubleshooting was not provided.
